lnbits/core/models.py

```diff
@@ -170,6 +178,13 @@ async def check_status(
 
         logger.debug(f"Status: {status}")
 
+        if self.is_in and status.pending and self.is_expired:
+            expiration_date = datetime.datetime.fromtimestamp(self.expiry)
+            logger.debug(
+                f"Deleting expired incoming payment {self.checking_id}: expired {expiration_date}"
+            )
+            await self.delete(conn)
```

Comment:
Check the expiry every time an invoice is checked (usually called by the frontend).

Reply:
just asking, because we had this issue recently, what happens to stuck outgoing invoices?

Reply:
Incoming invoices with a passed expiry will not (should not) be accepted by the receiver node so it is (should be) safe to assume that we can delete these payments immediately when the expiry is reached.